src/dawn/CodeGen/Cuda/MSCodeGen.hpp

```diff
@@ -54,11 +55,12 @@ class MSCodeGen {
   std::string cudaKernelName_;
   Array3ui blockSize_;
   const bool solveKLoopInParallel_;
+  CudaCodeGen::CudaCodeGenOptions& options_;
```

Comment:
const& or maybe not even a reference?

Reply:
I clearly vote for no reference as this object is rather small and it's just the configuration

Reply:
until when do we determine objects small enough to copy?

Reply:
Actually I need to correct myself, I only read the name of the variable and I thought this is the main options object. In that case it makes sense to do the copy because we want to look at the codegen part individually (and there it's done anyway).
In this case here, the usage is pretty simple (`MSCodeGen{options}.generate(...);`), so the object goes out of scope immediately. So that would be (and I know that I am contradicting what I wrote in the morning) an ideal candidate for `const&`. I don't think that it costs a lot to do the copy, because the string will be "small-string optimized" anyway (string=3*64bit), so it will only generate some boiler plate code for the unused copy.
So after carefully thinking about it again, I don't care that much; it is not critical code at all (regarding performance). Regarding pointer safety, it doesn't look critical either.

Reply:
Hannes and I had a quick discussion on this yesterday. His stance was that we actually want a copy of the object since the reference (const or not) means I / someone can change the options during code-gen. This we can be sure will not happen if we copy the object. So I think if we have a object where we do not want changes by someone else that is small enough (tbd) to copy we can go for that.

Thoughts on that?